RUMM-387 Add troubleshooting for handling core-ktx module conflicts (#227)
DataDog/dd-sdk-android: docs/TROUBLESHOOTING.md

This issue can occur if your dependencies rely on different Guava artifacts as the Datadog SDK for Android. The SDK uses AndroidX's WorkManager, which depends on a specific Guava dependency.

> Solve this issue by excluding the conflicting module from your dependency, eg:

Reviewer: ```suggestion
Solve this issue by excluding the conflicting module from your dependency, for example:
```

    jvmTarget = "1.8"
}
```
### Crash with `java.lang.NoSuchMethodError: No static method wrap(Landroid/view/WindowInsets)` in class `Landroidx/core/view/WindowInsetsCompat`

> Datadog SDK is using the dependency `androidx.core:core-ktx`. If you are also using this dependency, the build tools may have conflicts. For example, your compiled app could be using classes from version X and Y of `androidx.core:core-ktx` in which some methods called from X classes may no longer exist. The workaround is to drop the `androidx.core:core-ktx` transitive dependency when adding our SDK.

Reviewer 2: > For example, your compiled app could be using classes from version X and Y of `androidx.core:core-ktx` in which some methods called from X classes may no longer exist.

This is misleading and slightly confusing. I think the main issue is that their app is built against an older version of the core-ktx, and the method they expect is not available in the newest version. 

Dropping our library's dependency means that we might be using a method that's  not available yet on the older lib version, meaning that the build will still fail.

Author: it is not exactly that....the problem I experienced with the GoogleIO app was that they were using `core-ktx` version 1.1.0-alpha (greater than our version). For some reason the class loader loaded part of the classes from the old package and parts from the new package and there was a call to a function that was introduced in the new package which it was not able to find it (because in this case the class was loaded from the previous package)

Reviewer 2: How can the classloader use implementation from two different library version ? In theory the gradle dependency will always use the latest one for the whole project. 

But anyway the result is the sam, as i said, they expected a method that wasn't there. 

I think detailing the why and how of what the classloader does is not relevant here. dropping our dependency just hides the problem and can still generate another. My advice would be for them to upgrade their dependency version to make sure that the calls are compatible.

Author: Yeah...this was a mystery for me too as it cannot be possible for a ClassLoader to do that so we went back to the dependencies and looked deeper. Actually what is happening is that they are using a DrawerLayout class which is coming from `1.1.0 - alpha02`  and that class is calling a method inside the WindowInsetsCompat which is there in `core-ktx:1.1.0 - alpha02`. What happens and what I could not see last time is that our API brings in `core-ktx:1.2.0` and therefore gradle is picking this one and that method doesn't exist anymore there. So what you said is actually making sense that they should update to the latest version. Now question is, should we still add a troubleshooting for this case or we should assume that this is an obvious way of handling these issues.

Reviewer 2: I think we should not include it. We have a lot of (transitive) dependencies, so we can't troubleshoot every single incompatibility that can occur. 
People will raise issues here if they can't solve it on their own, and we'll help them case by case.

> If you are using the `androidx.core:core-ktx` have in mind that Datadog SDK is also using this dependency and the build tools sometimes are handling badly the dependencies conflicts. In short you can end up with a compiled app which is using classes from version X of `androidx.core:core-ktx` and some classes from version Y of `androidx.core:core-ktx` in which some methods called from X classes may no longer exists. The workaround is to drop the `androidx.core:core-ktx` transitive dependency when adding our SDK.

Reviewer: ```suggestion
Datadog SDK is using the dependency `androidx.core:core-ktx`. If you are also using this dependency, the build tools may have conflicts. For example, your compiled app could be using classes from version X and Y of `androidx.core:core-ktx` in which some methods called from X classes may no longer exist. The workaround is to drop the `androidx.core:core-ktx` transitive dependency when adding our SDK.
```

### Logs/Traces are not appearing in your dashboard.

> Make sure that you initialized the SDK using a valid [Client Token](https://docs.datadoghq.com/account_management/api-app-keys/#client-tokens).

Reviewer: ```suggestion
Initialize the SDK using a valid [Client Token](https://docs.datadoghq.com/account_management/api-app-keys/#client-tokens).
```
